extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/dashboard/services/DashboardDataFetcher.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (c) 2022 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.ext.wrapper.api.ui.pages.dashboard.services;
+
+import de.sovity.edc.ext.wrapper.api.ServiceException;
+import lombok.RequiredArgsConstructor;
+import org.eclipse.edc.connector.contract.spi.negotiation.store.ContractNegotiationStore;
+import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiation;
+import org.eclipse.edc.connector.policy.spi.PolicyDefinition;
+import org.eclipse.edc.connector.spi.contractagreement.ContractAgreementService;
+import org.eclipse.edc.connector.spi.policydefinition.PolicyDefinitionService;
+import org.eclipse.edc.connector.spi.transferprocess.TransferProcessService;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcess;
+import org.eclipse.edc.policy.model.Policy;
+import org.eclipse.edc.spi.asset.AssetIndex;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.edc.spi.types.domain.asset.Asset;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class DashboardDataFetcher {
+    private final ContractAgreementService contractAgreementService;
+    private final ContractNegotiationStore contractNegotiationStore;
+    private final TransferProcessService transferProcessService;
+    private final AssetIndex assetIndex;
+    private final PolicyDefinitionService policyDefinitionService;
+
+    public List<Asset> getAllAssets() {
+        return assetIndex.queryAssets(QuerySpec.max()).toList();
+    }
+
+    public List<Policy> getAllPolicies() {
```

Comment:
does the dashboard service really need the objects? same for other requests. where you just need the count you should only return the count here.